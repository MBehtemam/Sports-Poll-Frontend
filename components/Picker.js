const Picker = ({ items, onChange, defaultValue, keyValue, itemValue }) => (
  <select
    onChange={e => onChange(e.target.value)}
    defaultValue={defaultValue || ""}
  >
    <option key='srandom-10-1' diabled='true' value='Select'>
      ------------
    </option>
    {items.map(item => (
      <option key={item[keyValue]} value={item[keyValue]}>
        {item[itemValue]}
      </option>
    ))}
  </select>
);
export default Picker;
