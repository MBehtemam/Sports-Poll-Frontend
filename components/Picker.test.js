import React from "react";
import styled from "styled-components";
import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";
import "jest-styled-components";

import Picker from "./Picker";
const items = [
  { name: "name1", id: 1 },
  { name: "name2", id: 2 },
  { name: "name3", id: 3 }
];
describe("Test suits for Picker", () => {
  it("should match with snapshot", () => {
    const tree = renderer.create(
      <Picker items={items} keyValue='name' idValue='id' />
    );
    expect(tree).toMatchSnapshot();
  });
  it("should has correct children lenth", () => {
    const tree = shallow(<Picker items={items} keyValue='name' idValue='id' />);
  });
});
