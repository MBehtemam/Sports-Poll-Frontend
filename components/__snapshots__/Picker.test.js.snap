// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for Picker should match with snapshot 1`] = `
<select
  defaultValue=""
  onChange={[Function]}
>
  <option
    disabled={true}
    value=""
  >
    ------------
  </option>
  <option
    value="name1"
  />
  <option
    value="name2"
  />
  <option
    value="name3"
  />
</select>
`;
